app/clients/cloudwatch/aws_cloudwatch.py

```diff
@@ -2,12 +2,15 @@
 import os
 import re
 import time
+from datetime import datetime
 
+from _datetime import timedelta
```

Comment:
This import caught my eye; shouldn't `timedelta` just be included above after `datetime`?  Otherwise we're relying on underlying internal Python modules that the public libraries will already leverage properly.